spec/compiler/semantic/named_tuple_spec.cr

```diff
@@ -302,4 +302,23 @@ describe "Semantic: named tuples" do
       meta
     end
   end
+
+  it "doesn't crash on named tuple in not executed block (#6718)" do
+    assert_type(%(
+      require "prelude"
+
+      def pending(&block)
+      end
+
+      def untyped(x = nil)
+      end
+
+      # To reproduce this bug, it is needed to the expression that is
```

Comment:
FYI the correct english is "... it is needed to have an expression that is ..." and it reads even better as "... it is required to have an expression that is ..."